### PR TITLE
Adjust backend translations for entity components

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

Adjusts the structure of the backend translations for entity components (the fallback).

It also adds the fallback for attributes & state computations to the device class first.

Core PR: https://github.com/home-assistant/core/pull/89702
Dev docs: https://github.com/home-assistant/developers.home-assistant/pull/1714

TODO:
 - ~~Maybe just rename `state_attributes` to `attribute`.~~ Breaking/inconsistent, Let's not.
 - [x] Adjust loading of languages on connect
 - ~~Humidifier mode is extracted from backend translations~~ Those are not backend, so not a concern.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/1714

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/hui-thermostat-card.ts
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-thermostat-card.ts:483]:
```suggestion
        .label=${this.hass!.localize(
          `component.climate.entity_component._.state.${mode}`
        ) || mode}
```

**Review by piitaya — APPROVED** [review]
Can be merged once back-end is merged

**Comment by frenck** [comment]
Thanks, @piitaya ❤️

**Comment by frenck** [comment]
Backend has been merged.

**Comment by yangqian** [comment]
This commit breaks the translation for many languages. See https://github.com/home-assistant/core/issues/90957

**Pull request merged**
